CONTRIBUTING.md

```diff
@@ -1,3 +1,540 @@
+$ pipenv install --dev --skip-lock
+```
+
+Now you can verify we have installed a lot of stuff by running:
+
+```
+$ pip freeze
+```
+
+### Linking the SyMPC src
+
+Now we need to link the src directory of the SyMPC code base into our site-packages
+so that it acts like it’s installed but we can change any file we like and `import` again
+to see the changes.
+
+```
+$ pip install -e .
+```
+
+The best way to know everything is working is to run the tests.
+
+Run the quick tests with all your CPU cores by running:
+
+```
+$ pytest -n auto
+```
+
+If they pass then you know everything is set up correctly.
+
+
+### Linking the PySyft src
+
+SyMPC is a companion library for PySyft. For the moment they are highly coupled.
+One of the SyMPC goals is to convert it into a standalone library. Thus, PySyft
+will depend on SyMPC, but not the other way around. This point is not yet reached
+and this adds some complications when applying some changes that in SyMPC that requires changes in PySyft to work.
+
+**Example**: You implement a new function on a ShareTensor in SyMPC. In your test, you call that function from a PySyft object pointer `share_pointer_in_pysyft.new_method()`.  Clearly, this test will not work until PySyft uses this new SyMPC version you are developing.
+
+To test this, you will need to clone the PySyft library, install the dependencies and link the PySyft src as we have done with SyMPC. For more detailed information, please visit the [PySyft contributin guide](https://github.com/OpenMined/PySyft/blob/dev/packages/syft/CONTRIBUTING.md).
+
+As a little summary, you need to:
+
+```bash
+$ git clone https://github.com/OpenMined/PySyft.git
+$ cd PySyft/packages
+$ pip install -r 'requirements.txt'
+$ pip install -e syft
+```
+
+This steps are here as a guide. Please, check the [PySyft contributin guide](https://github.com/OpenMined/PySyft/blob/dev/packages/syft/CONTRIBUTING.md) for the detailed steps.
+
+Now, you are prepared to face this kind of changes. A normal change in SyMPC looks like:
+
+1. Change SyMPC code
+2. Run the tests
+
+A change that implies both libraries looks like:
+
+1. Change SyMPC code
+2. Change PySyft code
+3. Install SyMPC with `pip install -e .`
+4. Install PySyft with `pip install -e syft`
+5. Run SyMPC tests
+6. Run PySyft tests
+
+## Code Quality
+
+### Formatting and Linting
+
+We use several tools to keep our codebase high quality.
+
+- black
+- flake8
+- isort
+
+### Tests and CI
+
+When you push your code it will run through a series of GitHub Actions which will ensure that the code meets our minimum standards of code quality before a Pull Request can be reviewed and approved.
+
+To make sure your code will pass these CI checks before you push you should use the pre-commit hooks and run tests locally.
+
+We aim to have a 100% test coverage, and the GitHub Actions CI will fail if the coverage is below a certain value. You can evaluate your coverage using the following commands.
+
+```
+$ pytest -m fast -n auto
```

Comment:
NITPICK: I think you should run only `pytest -n auto tests` (We do not have slow vs fast tests -- this is for Syft to have faster local development)